### PR TITLE
Purchase Redirect: Consider active jetpack plugins when redirecting

### PR DESCRIPTION
**Pull request by jwebbdev**
#### Changes proposed in this Pull Request

* Instead of redirecting only to Jetpack, checks for jetpack-backup.
* We'll likely need a more general solution for other stand-alone plugins when they appear
* This also updates the WP Admin sidebar link in Calypso Green to redirect to Jetpack Backup if it's installed and Jetpack-the-plugin is not.

#### Testing instructions

* Install the Backup Plugin on a site
* Visit jetpack.com/upgrade/backup/?site=<your domain>
* i.e - https://jetpack.com/upgrade/backup/?site=www.example.com
* Add a plan to your cart
* Change the url to your calypso instance
* i.e. - http://calypso.localhost:3000/checkout/www.example.com#step2
* Checkout and ensure it redirects to the backup plugin
* Check the sidebar at calypso.localhost:3000 to ensure the WP Admin link points to jetpack-backup / jetpack accordingly
* Repeat with the Jetpack plugin, and ensure it redirects to the recommendations page as normal
* If repeated with both plugins installed all steps should default to Jetpack links, not Jetpack Backup

### CODE REVIEW

**Review by vianasw — APPROVED** [review]
Looks good to me.

File: client/my-sites/checkout/composite-checkout/hooks/use-get-thank-you-url/get-thank-you-page-url.ts [hunk 1]
```diff
@@ -312,12 +317,14 @@ function getFallbackDestination( {
 		if ( isJetpackNotAtomic && purchasedProduct ) {
 			debug( 'the site is jetpack and bought a jetpack product', siteSlug, purchasedProduct );
 
+			const adminPath = adminPageRedirect || 'admin.php?page=jetpack#/recommendations';
```
Comment (sirbrillig) [inline client/my-sites/checkout/composite-checkout/hooks/use-get-thank-you-url/get-thank-you-page-url.ts:320]: Could you also edit the tests in [this file](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/checkout/composite-checkout/test/composite-checkout-thank-you.js) to cover this modification?

**Comment by sirbrillig** [comment]
Thanks for adding the test!

**Comment by vianasw** [comment]
Tested again with the latest changes and redirect works perfectly.

**Pull request merged**